graph.go

```diff
@@ -718,3 +720,238 @@ func (g *Graph) validateTransactionParents(tx *Transaction) error {
 
 	return nil
 }
+
+func (g *Graph) processRewardWithdrawals(round uint64, snapshot *avl.Tree) {
+	rws := GetRewardWithdrawalRequests(snapshot, round-uint64(sys.RewardWithdrawalsRoundLimit))
+
+	for _, rw := range rws {
+		balance, _ := ReadAccountBalance(snapshot, rw.account)
+		WriteAccountBalance(snapshot, rw.account, balance+rw.amount)
+
+		snapshot.Delete(rw.Key())
+	}
+}
+
+func (g *Graph) RewardValidators(snapshot *avl.Tree, root *Transaction, tx *Transaction, logging bool) error {
```

Comment:
Why move validator logic into `graph.go`? Also, would be good to hide/expose it s.t. `rewardValidators`.

Reply:
To clarify, anything to do with rewarding/collapsing/staking should not belong in `graph.go`. Anything that modifies ledger state should remain in `ledger.go`, unless you have good reasons (say for testing infrastructure for example) to move it to `graph.go`.

Reply:
For testing, we should not need to initialize a full ledger instance just to test transaction processing.

Reply:
`graph.go` should be purely in-memory however and be entirely algorithmic. The functions moved to `graph.go` in this case modify ledger state.

`graph.go` already handles in-memory indices, managing the directed acyclic graph structure, and validating transactions. It's best not to have it handle actually applying transactions to a ledger state too.